File: clockMath.cpp

#include<bits/stdc++.h>
using namespace std;
 
int main()
{

int H,M;
cin>>H>>M;

double minuteAngle = M*6;
double hourAngle = (H%12)*30 + (M*.5);

double angle = abs(hourAngle - minuteAngle);

angle = min(angle, 360-angle);

cout<<fixed<<setprecision(4)<<angle<<endl;

    return 0;
}

File: clockMath.swift

import Foundation

if let input = readLine(){
    let values = input.split(separator: " ").compactMap{Int($0)}

    if values.count == 2{
        let H = values[0]
        let M = values[1]

        let minuteAngle = Double(M)*6
        let hourAngle = Double(H%12)*30 + Double(M)*0.5

        let angle = abs(hourAngle-minuteAngle)
        let smallerAngle = min(angle, 360-angle)

        print(String(format:"%.4f", smallerAngle))
    }
}
